odp_sdk/auth.py

```diff
@@ -58,6 +58,7 @@ class HardcodedTokenProvider(TokenProvider):
     """Token provider for hardcoded tokens"""
 
     def __init__(self, token: str):
+        super().__init__()
```

Comment:
`TokenProvider` is a subclass of `BaseModel`, so the token should instead be an attribute:

```python
class HardcodedTokenProvider(TokenProvider):
  token: PrivateAttr
```

Reply:
Like this?

```python
class HardcodedTokenProvider(TokenProvider):
    token: PrivateAttr(str)
    """Token provider for hardcoded tokens"""

    def get_token(self) -> str:
        return self._token

```